pkg/secrets/secrets.go

```diff
@@ -99,7 +99,65 @@ func generateUserDataWithPubKey(pubKey string) string {
 			$acl.SetAccessRule($administratorsRule)
 			$acl.SetAccessRule($systemRule)
 			$acl | Set-Acl
-			Restart-Service sshd`
+			Restart-Service sshd
+			`
+
+	if platformType == oconfig.AWSPlatformType {
+		userData = appendAwsUserDataConfig(userData)
+	}
+	return userData
+}
+
+// appendAwsUserDataConfig appends the AWS EC2Launch installation script to
+// the given userData to persist the instance metadata routes
+func appendAwsUserDataConfig(userData string) string {
+	// EC2Launch v2 minimum version where the task to persist routes was introduced
+	// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2launchv2-versions.html#ec2launchv2-version-history
+	EC2LaunchMinimumVersion := "2.0.1643"
+
+	// S3 URL for the latest EC2Launch version
+	// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2launch-v2-install.html#lv2-download-s3
+	EC2LaunchS3LatestURL := "https://s3.amazonaws.com/amazon-ec2launch-v2/windows/amd64/latest/AmazonEC2Launch.msi"
+
+	// append the EC2Launch installation script
+	userData += `function Install-LatestEC2LaunchV2 {
+				# set download dir
+				$DownloadDir = "$env:USERPROFILE\Desktop\EC2Launchv2"
+				New-Item -Path "$DownloadDir" -ItemType Directory -Force
+				# URL for your download location
+				$Url = "` + EC2LaunchS3LatestURL + `"
+				# set download location
+				$DownloadFile = "$DownloadDir" + "\" + $(Split-Path -Path $Url -Leaf)
+				# download the agent
+				Invoke-WebRequest -Uri $Url -OutFile $DownloadFile
```

Comment:
This may affect disconnected environments (air-gapped cluster on AWS infra). We can call out in our ocp docs that the EC2LaunchMinimumVersion should be 2.0.1643+

Reply:
good point, this logic will download the latest if
- no agent found or
- agent version is < `2.0.1643`

Reply:
happy to add a requirement in the existing disconnected docs https://github.com/openshift/windows-machine-config-operator/blob/master/README.md?plain=1#L234   

@openshift/openshift-team-windows-containers  WDYT?

Reply:
Pushed https://github.com/openshift/windows-machine-config-operator/pull/2296/commits/5c29c138b8447efce09b72b80f90aa3b4d3c9173 

@saifshaikh48   PTAL